ruff_lsp/server.py

```diff
@@ -1157,10 +1157,27 @@ async def _run_check_on_document(
     executable = _find_ruff_binary(settings, VERSION_REQUIREMENT_LINTER)
     argv: list[str] = CHECK_ARGS + list(extra_args)
 
+    skip_user_arg = False
     for arg in lint_args(settings):
+        if skip_user_arg:
+            skip_user_arg = False
+            continue
         if arg in UNSUPPORTED_CHECK_ARGS:
             log_to_output(f"Ignoring unsupported argument: {arg}")
         else:
+            if only:
+                if arg in (
+                    "--select",
+                    "--extend-select",
+                    "--ignore",
+                    "--extend-ignore",
+                ):
+                    skip_user_arg = True
+                    continue
+                elif arg.startswith(
```

Comment:
Why do you use `startswith` here? This check looks the same as the one above?

Reply:
For the following conditions:
1. `["--select", "A,B,C"]`
2. `["--select=A,B,C"]`

Thanks for checking, I'll add this as inline documentation